src/unix/glegl.cpp

```diff
@@ -0,0 +1,646 @@
+// ============================================================================
+// wxGLContext implementation
+// ============================================================================
+
+wxIMPLEMENT_CLASS(wxGLContext, wxObject);
+
+wxGLContext::wxGLContext(wxGLCanvas *win,
+                         const wxGLContext *other,
+                         const wxGLContextAttrs *ctxAttrs)
+    : m_glContext(NULL)
+{
+    const int* contextAttribs = NULL;
+
+    if ( ctxAttrs )
+    {
+        contextAttribs = ctxAttrs->GetGLAttrs();
+    }
+    else if ( win->GetGLCTXAttrs().GetGLAttrs() )
+    {
+        // If OpenGL context parameters were set at wxGLCanvas ctor, get them now
+        contextAttribs = win->GetGLCTXAttrs().GetGLAttrs();
+    }
+
+    m_isOk = false;
+
+    EGLConfig *fbc = win->GetEGLConfig();
+    wxCHECK_RET( fbc, "Invalid EGLConfig for OpenGL" );
+
+    m_glContext = eglCreateContext(wxGLCanvasEGL::GetDisplay(), fbc[0],
+                                   other ? other->m_glContext : EGL_NO_CONTEXT,
+                                   contextAttribs);
+
+    if ( !m_glContext )
+        wxLogMessage(_("Couldn't create OpenGL context"));
+    else
+        m_isOk = true;
+}
+
+wxGLContext::~wxGLContext()
+{
+    if ( !m_glContext )
+        return;
+
+    if ( m_glContext == eglGetCurrentContext() )
+        eglMakeCurrent(wxGLCanvasEGL::GetDisplay(), EGL_NO_SURFACE,
+                       EGL_NO_SURFACE, EGL_NO_CONTEXT);
+
+    eglDestroyContext(wxGLCanvasEGL::GetDisplay(), m_glContext);
+}
+
+bool wxGLContext::SetCurrent(const wxGLCanvas& win) const
+{
+    if ( !m_glContext )
+        return false;
+
+    return eglMakeCurrent(win.GetEGLDisplay(), win.GetEGLSurface(),
+                          win.GetEGLSurface(), m_glContext);
+}
+
+// ============================================================================
+// wxGLCanvasEGL implementation
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// initialization methods and dtor
+// ----------------------------------------------------------------------------
+
+wxGLCanvasEGL::wxGLCanvasEGL()
+{
+    m_config = NULL;
+    m_display = NULL;
+    m_surface = EGL_NO_SURFACE;
+    m_wl_compositor = NULL;
+    m_wl_subcompositor = NULL;
+    m_wl_frame_callback_handler = NULL;
+    m_wl_egl_window = NULL;
+    m_wl_surface = NULL;
+    m_wl_region = NULL;
+    m_wl_subsurface = NULL;
+    m_ready_to_draw = false;
+}
+
+bool wxGLCanvasEGL::InitVisual(const wxGLAttributes& dispAttrs)
+{
+    bool ret = InitConfig(dispAttrs, &m_config);
+    if ( !ret )
+    {
+        wxFAIL_MSG("Failed to get an EGLConfig for the requested attributes.");
+    }
+    return ret;
+}
+
+/* static */
+EGLDisplay wxGLCanvasEGL::GetDisplay()
+{
+    wxDisplayType type = wxDisplayNone;
+    void *display = wxGetDisplayAndType(&type);
+    EGLenum platform;
+    if (type == wxDisplayX11)
```

Comment:
When using enum it's preferable to use `switch` as the compiler can warn then when new enum values are added, but the code is not updated to handle them.